### PR TITLE
Add users, course and assessment to seed data

### LINKED ISSUE #1562: Create seed data for development
Enhance `db/seeds/development.rb` to add some useful data for local testing.
- [ ] Create a course
- [ ] Create some test users and assign them to the course
- [ ] Create an assessment and some questions
- [ ] Any other useful data

### PR DESCRIPTION
**Pull request by jeremyyap**
Add rake task to seed course and assessment data.

Courses: opened, published, closed
Assessments: published, published & autograded, unpublished
Users: students, teaching assistants, managers

Execute rake task with `bundle exec rake coursemology:seed`

Closes #1562

### CODE REVIEW

**Comment by fonglh** [comment]
There's a manager role also.

https://github.com/Coursemology/coursemology2/blob/master/app/models/course_user.rb#L13

**Comment by weiqingtoh** [comment]
Should we put it in the seed file or a separate rake task?

**Comment by kxmbrian** [comment]
Should we be using FactoryGirl for this? Maybe throw in the 'faker' gem?

@jeremyyap you [can set `User.stamper`](https://github.com/magiclabs/userstamp).

**Comment by jeremyyap** [comment]
Trying to use the FactoryGirl factories in `spec/factories`, but using the `user` factory results in long email addresses with unix timestamp included e.g. `user_1@domain_1481451340_name.com` due to a sequence defined in the `user_email` factory.

File: db/seeds/coursemology_dev.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+require 'factory_girl_rails'
+require 'coursemology/polyglot'
+
+ActsAsTenant.with_tenant(Instance.default) do
+  # Set the stamper for creator and updater fields
+  admin = User::Email.find_by_email('test@example.org').user
+  User.stamper = admin
+
+  # Create courses
+  open_course = FactoryGirl.create(:course, :opened)
```
Comment (weiqingtoh) [inline db/seeds/coursemology_dev.rb]: Sorry, have a few more suggestions:
1. Let `admin` be owner of this course
2. Let `admin` be `course_staff` of another course (maybe `published_course`)
3. Add some indication to the course title (similar to the assessment titles)

File: db/seeds/coursemology_dev.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  # Add users to courses
+  courses.each do |course|
+    20.times { FactoryGirl.create(:course_student, course: course) }
+    5.times { FactoryGirl.create(:course_teaching_assistant, course: course) }
+    5.times { FactoryGirl.create(:course_manager, course: course) }
+  end
+
+  # Add assessments
+  courses.each do |course|
```
Comment (weiqingtoh) [inline db/seeds/coursemology_dev.rb]: Actually why not just add the users and assessments in the same loop?

File: spec/coverage_helper.rb [hunk 3]
```diff
@@ -59,5 +59,8 @@ def expand_formatter(formatter)
     # Extra statistics to be placed in `rake stats`. We don't run that on CI, so coverage is not
     # important.
     add_filter '/lib/tasks/coursemology/stats_setup.rake'
+
+    # Rake task to seed dev database with course and assessment data.
+    add_filter '/lib/tasks/coursemology/seed.rake'
```
Comment (weiqingtoh) [inline spec/coverage_helper.rb]: Don't think we need this right? No point seeding the travis database as we create our own objects for different tests.

Comment (jeremyyap) [inline spec/coverage_helper.rb]: This is to exclude it from test coverage calculation

Comment (weiqingtoh) [inline spec/coverage_helper.rb]: Ohhhhhhh 👍

**Comment by weiqingtoh** [comment]
LGTM. The task `rake coursemology:seed` would also ensure that `db:seed` is run - over time when we have more complex cases, it might be helpful to update this again.

**Comment by weiqingtoh** [comment]
Thoughts @allenwq @fonglh @kxmbrian ?

**Comment by fonglh** [comment]
Is the programming question created a real one?

**Comment by weiqingtoh** [comment]
@fonglh I suspect not, unless we create an actual programming package question? Maybe a hello world?

**Comment by allenwq** [comment]
We must run `db:seed` before `coursemology:seed` or ? I am confused

File: lib/tasks/coursemology/seed.rake [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+                                          name: 'Administrator')
+
+      # Other courses are owned by admin user
+      User.stamper = admin
+      open_course = FactoryGirl.create(:course, :opened, title: 'Open Course')
+      closed_course = FactoryGirl.create(:course, :closed, title: 'Closed Course')
+
+      [open_course, published_course, closed_course].each do |course|
+        # Add users to courses
+        20.times { FactoryGirl.create(:course_student, course: course) }
```
Comment (kxmbrian) [inline lib/tasks/coursemology/seed.rake]: Better to use `FactoryGirl.create_list` instead?

**Comment by fonglh** [comment]
There're some samples here.
https://github.com/coursemology/evaluator-samples

**Comment by jeremyyap** [comment]
I added an assessment with an actual programming question same as the one at https://github.com/coursemology/evaluator-samples i.e. return `True` to pass, `False` to fail

Not sure where to park the zip file, currently it is at `lib/tasks/coursemology/programming_question.zip`

**Comment by allenwq** [comment]
@jeremyyap Have tested whether the seeds works with the latest code base? 

Let's get this merged first.

**Comment by jeremyyap** [comment]
Updated seed data for course `enrollable` boolean. Seeding task works on latest master.

**Pull request merged**